### PR TITLE
Makefile : generate protobuf cpp files at build

### PR DESCRIPTION
**Pull request by shjala**
Generate protobuf at build time, static ones might result in compilation/linkage error due to calling changed private protobuf functions, if newer protobuf library is used.

### CODE REVIEW

**Comment by shjala** [comment]
@eriknordmark please review.

File: eve-tools/Makefile [hunk 1]
```diff
@@ -11,15 +11,18 @@ DESTDIR = /usr/
 PREFIX = local
 
 eve_run: libevetools.so
-	$(CC) tools/eve_run.cpp cpp/api.pb.cc $(CPPFLAGS) -o eve_run $(LDLIBS)
+	$(CC) tools/eve_run.cpp proto/api.pb.cc $(CPPFLAGS) -o eve_run $(LDLIBS)
 
-libevetools.so: lib/src/sendrecv.cpp
-	$(CC) $(CPPFLAGS) -shared -fPIC -o libevetools.so lib/src/sendrecv.cpp lib/src/eve_tpm_service.cpp cpp/api.pb.cc -lprotobuf
+protoc_files:
+	protoc --cpp_out=proto -Iproto api.proto
```
Comment (eriknordmark) [inline eve-tools/Makefile:17]: Should we document in README.md that protoc needs to be installed to build this?

Comment (shjala) [reply eve-tools/Makefile:17]: It is listed in the [INSTALL.md](https://github.com/lf-edge/eve-tools/blob/master/INSTALL.md#1-install-the-prerequisite-libraries), the `protobuf-compiler`.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Comment by shjala** [comment]
@cshari-zededa gentle reminder to review.

**Review by cshari-zededa — APPROVED** [review]
LGTM.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**